### PR TITLE
Added a test to verify user agent prefix can be set correctly.

### PR DESCRIPTION
**Pull request by makdharma**

### CODE REVIEW

File: src/objective-c/tests/GRPCClientTests.m [hunk 1]
```diff
@@ -257,6 +259,36 @@ - (void)testResponseMetadataKVO {
...
+- (void)testUserAgentPrefix {
+  __weak XCTestExpectation *response = [self expectationWithDescription:@"Empty response received."];
+  __weak XCTestExpectation *completion = [self expectationWithDescription:@"Empty RPC completed."];
+
+  GRPCCall *call = [[GRPCCall alloc] initWithHost:kHostAddress
+                                             path:kEmptyCallMethod.HTTPPath
+                                   requestsWriter:[GRXWriter writerWithValue:[NSData data]]];
+  // Setting this special key in the header will cause the interop server to echo back the
+  // user-agent value, which we confirm.
+  call.requestHeaders[@"x-grpc-test-echo-useragent"] = @"";
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: Interesting! A header without value is valid HTTP/2?

File: src/objective-c/tests/GRPCClientTests.m [hunk 2]
```diff
@@ -257,6 +259,36 @@ - (void)testResponseMetadataKVO {
...
+
+  id<GRXWriteable> responsesWriteable = [[GRXWriteable alloc] initWithValueHandler:^(NSData *value) {
+    XCTAssertNotNil(value, @"nil value received as response.");
+    XCTAssertEqual([value length], 0, @"Non-empty response received: %@", value);
+    XCTAssertEqualObjects(call.responseHeaders[@"x-grpc-test-echo-useragent"],
+                    @"Foo grpc-objc/0.13.0 grpc-c/0.14.0-dev (ios)",
+                    @"Did not receive expected user agent %@",
+                    call.responseHeaders[@"x-grpc-test-echo-useragent"]);
+
+      [response fulfill];
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: nit: indentation

File: src/objective-c/tests/GRPCClientTests.m [hunk 3]
```diff
@@ -257,6 +259,36 @@ - (void)testResponseMetadataKVO {
...
+                                   requestsWriter:[GRXWriter writerWithValue:[NSData data]]];
+  // Setting this special key in the header will cause the interop server to echo back the
+  // user-agent value, which we confirm.
+  call.requestHeaders[@"x-grpc-test-echo-useragent"] = @"";
+
+  id<GRXWriteable> responsesWriteable = [[GRXWriteable alloc] initWithValueHandler:^(NSData *value) {
+    XCTAssertNotNil(value, @"nil value received as response.");
+    XCTAssertEqual([value length], 0, @"Non-empty response received: %@", value);
+    XCTAssertEqualObjects(call.responseHeaders[@"x-grpc-test-echo-useragent"],
+                    @"Foo grpc-objc/0.13.0 grpc-c/0.14.0-dev (ios)",
```
Comment (jcanizales) [inline src/objective-c/tests/GRPCClientTests.m]: nit: either indent 4 spaces to continue the line (go left), or align with the first argument (go right).

**Comment by jcanizales** [comment]
Awesome! Just two style nits and a question.

File: src/objective-c/tests/GRPCClientTests.m (hunk 1 above)
Comment (makdharma) [inline src/objective-c/tests/GRPCClientTests.m]: apparently.. I'm not able to find a reference that says otherwise.

**Comment by jcanizales** [comment]
Test failures are unrelated. Merging.

**Pull request merged**